sdk/core/scheduler/thread.h

```diff
@@ -543,6 +539,17 @@ namespace
 			return next != this;
 		}
 
+		/**
+		 * Returns true if the thread has run for a complete tick.  This must
+		 * be called only on the currently running thread.
+		 */
+		bool has_run_for_full_tick()
+		{
+			Debug::Assert(this == current,
+			              "Only the current thread is running");
```

Comment:
That's a bit of a tautology, isn't it?

```suggestion
			              "Ask only about currently running thread");
```

Reply:
Wes' version is verging on a parody of an ancient Chinese wisdom. "Ask not a suspended thread how long it has been running for."

Reply:
May you be scheduled at interesting times.